src/core/interfaces.ts

```diff
@@ -46,4 +46,12 @@ module Plottable {
     topLeft: Point;
     bottomRight: Point;
   };
+
+  export interface Entity<C extends Component> {
+    datum: any;
+    index: number;
+    position: Point;
+    selection: d3.Selection<any>;
+    component: C;
```

Comment:
Location is better than position, so I am okay with that change alone. 

Other than that, I have the feeling that this object is too verbose (exposing too much). Is everything above strictly necessary? This actually now brings us back to the other thread (above)